### PR TITLE
document enhance, and removed depreciate package

### PR DESCRIPTION
**Pull request by salmanhasni**
**ISSUES:**
1-gcloud-node official pacakge is now rename to google-cloud-node
2-No need to make file public (get file with bucket authentication and pipe it to meteor application user)

### CODE REVIEW

**Comment by salmanhasni** [comment]
@dr-dimitru Can you please review this

File: docs/google-cloud-storage-integration.md [hunk 1]
```diff
@@ -100,15 +96,13 @@ Collections.files = new FilesCollection({
...
     if (path) {
       // If file is moved to Google Cloud Storage
       // We will pipe request to Google Cloud Storage
       // So, original link will stay always secure
-      Request({
-        url: path,
-        headers: _.pick(http.request.headers, 'range', 'accept-language', 'accept', 'cache-control', 'pragma', 'connection', 'upgrade-insecure-requests', 'user-agent')
-      }).pipe(http.response);
+      var remoteReadStream = bucket.file(path).createReadStream();
+      remoteReadStream.pipe(http.response);
```
Comment (dr-dimitru) [inline docs/google-cloud-storage-integration.md]: @salmanhasni 
Have you tested this with video/audio files, is time seeking working?

Comment (salmanhasni) [inline docs/google-cloud-storage-integration.md]: @dr-dimitru 
tested it with static files (images, docs)
need to test it with audio/video files for time seeking.

I will let you know once i will it.

Comment (dr-dimitru) [inline docs/google-cloud-storage-integration.md]: @salmanhasni waiting for test results

So you need test results for:
 - Standart request for static and streaming files (seeking may or may not work)
 - Download (with `?download=true` query) for static and streaming files (browser should show remaining time and speed)
 - Streaming (with `?play=true` query) for static and streaming files (time seeking __must__ work)

Comment (dr-dimitru) [inline docs/google-cloud-storage-integration.md]: @salmanhasni any news on your end?

Comment (salmanhasni) [inline docs/google-cloud-storage-integration.md]: @dr-dimitru sorry for the delay response, i was out for vacations. 
I have test static (images etc) and dynamic files ( video, audio), 
Standard request and download request works fine, where as streaming video having time seeking issue.

Comment (dr-dimitru) [inline docs/google-cloud-storage-integration.md]: You can make it work (*I think so*) by passing `readableStream` to [`.serve()`](https://github.com/VeliovGroup/Meteor-Files/blob/master/files.coffee#L2218) method.
Could you try this?

Comment (salmanhasni) [inline docs/google-cloud-storage-integration.md]: @dr-dimitru I am able to make this PR work, but for that I need to write the below redundant code for making stream videos work with time seeking.
Should i refactor it ? 

```
if http.request.headers.range
      partiral = true
      array    = http.request.headers.range.split /bytes=([0-9]*)-([0-9]*)/
      start    = parseInt array[1]
      end      = parseInt array[2]
      end      = vRef.size - 1 if isNaN(end)
      take     = end - start
    else
      start    = 0
      end      = vRef.size - 1
      take     = vRef.size

    if partiral or (http.params.query.play and http.params.query.play == 'true')
      reqRange = {start, end}
      if isNaN(start) and not isNaN end
        reqRange.start = end - take
        reqRange.end   = end
      if not isNaN(start) and isNaN end
        reqRange.start = start
        reqRange.end   = start + take

      reqRange.end = vRef.size - 1 if ((start + take) >= vRef.size)

      if self.strict and (reqRange.start >= (vRef.size - 1) or reqRange.end > (vRef.size - 1))
        responseType = '416'
      else
        responseType = '206'
    else
      responseType = '200'

     var remoteReadStream;

     if(responseType == "206") {
         remoteReadStream = bucket.file(path).createReadStream();
     }
     else f(responseType == "200"){
         remoteReadStream = bucket.file(path).createReadStream();
     }
     self.serve(http, fileRef, fileRef.versions[version], version, remoteReadStream);
```

Comment (dr-dimitru) [inline docs/google-cloud-storage-integration.md]: Just pass `remoteReadStream` as 5th argument to `serve()` method.
No need write more code, see this example: https://github.com/VeliovGroup/Meteor-Files/blob/master/demo/lib/files.collection.coffee#L86

Comment (dr-dimitru) [inline docs/google-cloud-storage-integration.md]: @salmanhasni looks like I'm wrong...
Is there a way to `.createReadStream` form `bucket` with `start` and `end`?

Comment (salmanhasni) [inline docs/google-cloud-storage-integration.md]: @dr-dimitru Yes we can.

Sorry for my side, I just forgot to add reqRange params in createReadStream method for responseType 206, please see the above update code.

Comment (dr-dimitru) [inline docs/google-cloud-storage-integration.md]: @salmanhasni 
Sounds like you made it work, right?
If so, does tests passed?
If tests is passed re-open this PR, and I'll merge it. :)

**Comment by salmanhasni** [comment]
@dr-dimitru 
All test passed,
we are good to proceed.

**Comment by dr-dimitru** [comment]
@salmanhasni great. Thank you for this.
I'll do test on my end, and merge this PR.

**Comment by salmanhasni** [comment]
@dr-dimitru Sure

**Pull request merged**